Add details to devtools release notes. (#7524)
flutter/website: src/development/tools/devtools/release-notes/release-notes-2.17.0-src.md

Dart & Flutter DevTools - A Suite of Performance Tools for Dart and Flutter

## Inspector Updates

> * Added support for manually setting the package directories for your app. If you've ever 
>   loaded the Inspector and noticed that some of your widgets are not present in the 
>   Widget tree, this may be an indication that the package directories for your app 
>   have not been set or detected properly. Your package directories determine which 
>   widgets the Inspector considers to be from "your" application. If you see an 
>   empty Inspector widget tree, or if you develop widgets across multiple packages 
>   and want widgets from all these locations to show up in your tree, check the 
>   Inspector Settings dialog to ensure that your package directories are properly
>   configured. - [#4306](https://github.com/flutter/devtools/pull/4306)

Reviewer: ```suggestion
* Added support for manually setting the package directories for your app.
  If you've ever loaded the Inspector and noticed that some of your widgets
  aren't present in the widget tree, this might indicate that the package directories
  for your app haven't been set or detected properly.
  Your package directories determine which widgets
  the Inspector considers to be from _your_ application.
  If you see an empty Inspector widget tree,
  or if you develop widgets across multiple packages 
  and want widgets from all these locations to show up in your tree,
  check the **Inspector Settings** dialog to ensure that your package
  directories are properly
  configured. - [#4306](https://github.com/flutter/devtools/pull/4306)
```

  
<img
  alt="package directories"
  src="{{site.url}}/development/tools/devtools/release-notes/images-2.17.0/package_directories.png"
  width="800"
/>

## Performance Updates

> * Added a Frame Analysis tab to the Performance page. When analyzing a janky Flutter frame,
>   this view provides hints for how to diagnose the jank and detects expensive operations
>   that may have contributed to the slow frame time. This view also shows a breakdown of
>   your Flutter frame time per phase (Build, Layout, Paint, and Raster) to try to guide
>   you in the right direction - [#4339](https://github.com/flutter/devtools/pull/4339)

Reviewer: ```suggestion
* Added a **Frame Analysis** tab to the Performance page.
  When analyzing a janky Flutter frame,
  this view provides hints for how to diagnose the jank and
  detects expensive operations that might have contributed
  to the slow frame time. This view also shows a breakdown of
  your Flutter frame time per phase (**Build**, **Layout**, **Paint**, and **Raster**)
  to try to guide you in the right direction
  - [#4339](https://github.com/flutter/devtools/pull/4339)
```

Dart & Flutter DevTools - A Suite of Performance Tools for Dart and Flutter

## Inspector Updates
* Added support for manually setting the package directories for your app. If you've ever 
  loaded the Inspector and noticed that some of your widgets are not present in the 
  Widget tree, this may be an indication that the package directories for your app 
  have not been set or detected properly. Your package directories determine which 
  widgets the Inspector considers to be from "your" application. If you see an 
  empty Inspector widget tree, or if you develop widgets across multiple packages 
  and want widgets from all these locations to show up in your tree, check the 
  Inspector Settings dialog to ensure that your package directories are properly

>   configured. 

Reviewer 2: link the PR with a single dash separator to match what we do for other release notes:

```
"... properly configured - [#4306](https://github.com/flutter/devtools/pull/4306)"
```

  
  [#4306](https://github.com/flutter/devtools/pull/4306)
  
<img
  alt="package_directories"
  src="{{site.url}}/development/tools/devtools/release-notes/images-2.17.0/package_directories.png"
  width="800"

> />

Reviewer 2: can you use the image format that is used in the other release notes?

e.g.
```
![auto-complete-suggestions]({{site.url}}/development/tools/devtools/release-notes/images-2.14.0/image3.png "auto complete suggestions")
```

Author: the screen shot is much wider than page

the only way to narrow it in code os to use plain html
or, I can reduce resolution of the image (with little lost of quality)

Reviewer 2: @johnpryan @sfshaza2 is there a best practice for image sizing?

Reviewer: To be honest, the best approach is to optimize the image to make it smaller. However, we do prefer that you use the markdown method for resizing images. You can do that as follows:

```
![auto-complete-suggestions]({{site.url}}/development/tools/devtools/release-notes/images-2.14.0/image3.png "auto complete suggestions"){:width="90%"}
```

## Performance Updates
* Added frame analysis that enables hints that:
  * Point users to the "Enhance Tracing" when a frame phase is the leading cause of slow UI time
  * Point users to the "Raster Metrics" tab for frames with raster jank
  * Point users to documentation when expensive operations like Canvas.saveLayer(), intrinsics calculations, and shader compilation are contributing to jank.
  * Provide a visual breakdown of where the time in a frame was spent.

  [#4339](https://github.com/flutter/devtools/pull/4339)
  

>   Example analysis view for a janky frame:

Reviewer 2: nit: I don't think we need this line. It is clear that this image is an example because it directly follows this Release Notes entry.

> ![frame_analysis]({{site.url}}/development/tools/devtools/release-notes/images-2.17.0/frame_analysis.png)

Reviewer 2: add the accessibility description at the end (see example above)
